src/EditorFeatures/Core.Wpf/Completion/Presentation/ToolTipProvider.cs

```diff
@@ -55,13 +60,12 @@ public UIElement GetUIElement(VSCompletion itemToRender, ICompletionSession cont
 
         private class CancellableContentControl : ContentControl
         {
-            private readonly ForegroundThreadAffinitizedObject _foregroundObject = new ForegroundThreadAffinitizedObject();
             private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
             private readonly ToolTipProvider _toolTipProvider;
 
             public CancellableContentControl(ToolTipProvider toolTipProvider, CustomCommitCompletion item)
             {
-                Debug.Assert(_foregroundObject.IsForeground());
+                Debug.Assert(toolTipProvider._threadingContext.JoinableTaskContext.IsOnMainThread);
```

Comment:
can we have an extension on IThreadingContext for this?

Reply:
➡️ Added to the follow-up items list (main thread check/assert)